### DISCUSSION TITLE
TinyMCE 6 - checkbox onChange event not being triggered on checking checkbox

### DISCUSSION THREAD

**Discussion by ashvin-bhuttoo** [tinymce/tinymce#9440] [General]
## 📝 Provide detailed reproduction steps (if any)
Please provide the steps to reproduce and if possible a minimal demo of the problem via [fiddle.tiny.cloud](fiddle.tiny.cloud) or similar.
1. [Codepen link](https://codepen.io/ashvinbhuttoo/pen/abxojra)
Click on custom button and toggle the checkbox

### ✔️ Expected result
The label customLabel should show up, a js alert should pop up with text "checkbox clicked"

### ❌ Actual result
The onChange function is not being triggered

## 📃 Other details
- Browser: Chrome latest
- OS: Windows 11
- First affected version: ?
- Worked in version: 6

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by lostkeys** [comment]
Hi @ashvin-bhuttoo. You should use the [URL dialog](https://www.tiny.cloud/docs/tinymce/6/urldialog/) to create this type of custom functionality. I'll convert this to a discussion topic as its not a bug with TinyMCE.

— Fredrik, Product Manager

**Reply by ashvin-bhuttoo** [reply]
In [tiny docs](https://www.tiny.cloud/docs/tinymce/latest/dialog-components/#checkbox) it is clearly mentionned:

> Events: Interacting with a checkbox component will fire the onChange function in the dialog’s configuration.

So this is actually a bug. but thanks anyway for pointing out the possible workaround.
